frontend/src/components/Playback/PlayerElement.vue

```diff
@@ -1,66 +1,180 @@
 
 const playbackManager = playbackManagerStore();
+const playerElement = playerElementStore();
 const { t } = useI18n();
 
-const mediaElement = computed<'audio' | 'video' | undefined>(() => {
-  if (playbackManager.currentlyPlayingMediaType === 'Audio') {
-    return 'audio';
-  } else if (playbackManager.currentlyPlayingMediaType === 'Video') {
-    return 'video';
-  }
-});
+let hls: Hls | undefined;
 
 /**
  * If the player is a video element and we're in the PiP player or fullscreen video playback, we need to ensure the DOM elements are mounted before the teleport target is ready
  */
-const teleportTarget = computed<'body' | '.video-container'>(() =>
-  mediaElement.value === 'audio' ? 'body' : '.video-container'
-);
+const teleportTarget = computed<
+  'body' | '.fullscreen-video-container' | '.minimized-video-container'
+>(() => {
+  if (playbackManager.currentlyPlayingMediaElement === 'audio') {
+    return 'body';
+  } else if (
+    playbackManager.currentlyPlayingMediaElement === 'video' &&
+    playerElement.isFullscreenMounted
+  ) {
+    return '.fullscreen-video-container';
+  } else if (
+    playbackManager.currentlyPlayingMediaElement === 'video' &&
+    playerElement.isPiPMounted
+  ) {
+    return '.minimized-video-container';
+  } else {
+    return 'body';
+  }
+});
 
 const posterUrl = computed<string>(() =>
-  !isNil(playbackManager.currentItem) && mediaElement.value === 'video'
+  !isNil(playbackManager.currentItem) &&
+  playbackManager.currentlyPlayingMediaElement === 'video'
     ? getImageInfo(playbackManager.currentItem, {
         preferBackdrop: true
       }).url || ''
     : ''
 );
 
+watch(
+  () => playbackManager.currentSubtitleStreamIndex,
+  () => playerElement.applyCurrentSubtitle()
+);
+
+/**
+ * Called by the media element when the playback is ready
+ */
+function onCanPlay(): void {
+  if (playbackManager.currentlyPlayingMediaType === 'Video') {
+    playerElement.applyCurrentSubtitle();
+  }
+}
+
+/**
+ * Callback for when HLS.js gets an error
+ */
+function onHlsEror(event: Events.ERROR, data: ErrorData): void {
+  if (data.fatal) {
+    switch (data.type) {
+      case ErrorTypes.NETWORK_ERROR: {
+        // try to recover network error
+        console.log('fatal network error encountered, try to recover');
+        hls?.startLoad();
+        break;
+      }
+      case ErrorTypes.MEDIA_ERROR: {
+        console.log('fatal media error encountered, try to recover');
+        hls?.recoverMediaError();
+        break;
+      }
+      default: {
+        // cannot recover
+        hls?.destroy();
+        break;
+      }
+    }
+  }
+}
+
 watch(
   () => playbackManager.currentSourceUrl,
-  () => {
-    if (
-      mediaElementRef.value &&
-      playbackManager.currentSourceUrl &&
-      playbackManager.currentMediaSource?.SupportsDirectPlay &&
-      playbackManager.currentlyPlayingMediaType === 'Audio'
-    ) {
-      mediaElementRef.value.src = playbackManager.currentSourceUrl;
+  async () => {
```

Comment:
Fix SonarCloud complexity issue here please. Some ideas:

* if ``currentlyPlayingMediaType === 'Audio'`` -> do all the logic for music -> else if ``currentlyPlayingMediaType === 'Video'`` -> do all the logic for video, which includes hls.
* Do a ``const isHlsSupported = Hls.isSupported()`` at the beginning of this component. I highly doubt the truthy value of the hls.js support for the browser will ever change during the lifecycle of the application.
* Do the same with a ``isNativeHlsSupported`` variable.

Reply:
Done